model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaModelCriteriaBuilder.java

```diff
@@ -104,4 +110,29 @@ public BiFunction<CriteriaBuilder, Root<E>, Predicate> getPredicateFunc() {
     public boolean isDistinct() {
         return this.isDistinct;
     }
+
+    @SuppressWarnings("unchecked")
+    protected Collection<?> getValuesForInOperator(Object[] values, SearchableModelField<?> modelField) {
+        if (values == null || values.length == 0) throw new CriterionNotSupportedException(modelField, Operator.IN);
+
+        Collection<?> collectionValues;
+        if (values.length == 1) {
+
+            if (values[0] instanceof Object[]) {
+                collectionValues = Arrays.asList(values[0]);
+            } else if (values[0] instanceof Collection) {
+                collectionValues = (Collection) values[0];
+            } else if (values[0] instanceof Stream) {
+                try (Stream<?> str = ((Stream) values[0])) {
+                    collectionValues = str.collect(Collectors.toCollection(ArrayList::new));
+                }
+            } else {
+                collectionValues = Collections.singleton(values[0]);
+            }
+
+        } else  {
+            collectionValues = new HashSet(Arrays.asList(values));
+        }
+        return collectionValues;
+    }
```

Comment:
@vramik I've borrowed the above code from your authz PR. Once that is merged I'll update.